ament_index_cpp/test/utest.cpp

```diff
@@ -266,3 +276,50 @@ TEST(AmentIndexCpp, get_packages_with_prefixes) {
   // baz is found in prefix 2 only
   EXPECT_EQ(generate_subfolder_path("prefix2"), packages_with_prefixes["baz"]);
 }
+
+TEST(AmentIndexCpp, has_empty_name){
+  EXPECT_THROW(
+      ament_index_cpp::has_resource("type", ""),
+      std::runtime_error);
+}
+
+TEST(AmentIndexCpp, has_empty_type){
+  EXPECT_THROW(
+      ament_index_cpp::has_resource("", ""),
```

Comment:
Consider:
```cpp
ament_index_cpp::has_resource("", "name")
```

In that way, you ensure it's failing because of empty type, and not empty name.